posts/2022-08-08-daytrader8-performance.adoc

```diff
@@ -45,49 +45,49 @@ Vijay Sundaresan <https://github.com/vijaysun-omr>
 
 The first metric we compared was the time to receive the first response back at the client (sometimes equated with startup time). Liberty was almost 80% better than the nearest competitor (JBoss EAP) and this was a testament not just to the EE8 application server technology in Liberty but also to the optimized containers that include a shared classes cache population step as part of the container build procedure.    
 
--------------------------
-[.img_border_light]
+//-------------------------
+//[.img_border_light]
 image::img/blog/dt8pic1.png[First Response Times ,width=70%,align="center"]
--------------------------
+//-------------------------
 
 The next metric compared was the memory footprint just after the first response was sent back to the client. This shows the memory consumed by the server framework itself, separate from the memory that will be used by the application as workload increases. Liberty takes less than half the memory as the nearest competitor, while still consistently outperforming on the other performance metrics.  
  
--------------------------
-[.img_border_light]
+//-------------------------
+//[.img_border_light]
 image::img/blog/dt8pic2.png[Memory Footprint ,width=70%,align="center"]
--------------------------
+//-------------------------
 
 Throughput is usually the metric many customers associate with “performance” since it measures the number of requests that the application server was able to handle in a fixed unit of time. As the below comparison shows, Liberty has a very well tuned Java EE8 implementation designed for both optimal peak throughput performance and judicious resource use. 
   
--------------------------
-[.img_border_light]
+//-------------------------
+//[.img_border_light]
 image::img/blog/dt8pic3.png[Throughput ,width=70%,align="center"]
--------------------------
+//-------------------------
 
 Finally, the last metric that we evaluated was the scalability (sometime also referred to as “scaling up”, since there is only a single server instance). In this evaluation we varied the number of simultaneous client requests (we were using JMeter to drive the load) so that we could get a view into the application server’s behavior at different load levels to assess if the above throughput differences were unique to a particular configuration. The scalability experiments used 2 CPUs overall and so it was relatively easy to saturate the CPU with a moderate number of client requests as can be seen by the throughput curves flattening out in the below comparison. Once again Liberty stands out for the superior throughput at many different load levels showing that it can scale up consistently as load increases (harder to judge at the lower load levels since it is likely that CPU was not saturated).
```

Comment:
Re " (harder to judge at the lower load levels since it is likely that CPU was not saturated)" - I've removed it because it doesn't seem to add much, slightly contradicts the previous statement, and softens the finding itself.

```suggestion
== Scales up with load increases to higher throughput than competitors

The final metric that we evaluated was "scaling up", the scalability of the server instance. In this evaluation we varied the number of simultaneous client requests (we were using JMeter to drive the load) so that we could get a view into the runtime’s behavior at different load levels to assess if the previously described throughput differences were unique to a particular configuration.

The scalability experiments used two CPUs overall, so it was relatively easy to saturate the CPU with a moderate number of client requests as can be seen by the throughput curves flattening out in the following chart. Once again Liberty stands out for the superior throughput at many different load levels showing that it can scale up consistently as load increases.
```